google/cloud/ndb/_datastore_query.py

```diff
@@ -801,10 +801,22 @@ def _compare(self, other):
             return NotImplemented
 
         for order in self.order_by:
-            this_value_pb = self.result_pb.entity.properties[order.name]
-            this_value = helpers._get_value_from_value_pb(this_value_pb)
-            other_value_pb = other.result_pb.entity.properties[order.name]
-            other_value = helpers._get_value_from_value_pb(other_value_pb)
+
+            if order.name == '__key__':
+                this_value = helpers.key_from_protobuf(self.result_pb.entity.key)
```

Comment:
I think you could probably just append `.flat_path` to the ends of these two lines and save the extra `if` statements below.